MonoGame.Framework/Color.cs

```diff
@@ -273,14 +288,43 @@ public Color(int r, int g, int b)
         public Color(int r, int g, int b, int alpha)
         {
             _packedValue = 0;
-            R = (byte)MathHelper.Clamp(r, Byte.MinValue, Byte.MaxValue);
-            G = (byte)MathHelper.Clamp(g, Byte.MinValue, Byte.MaxValue);
-            B = (byte)MathHelper.Clamp(b, Byte.MinValue, Byte.MaxValue);
-            A = (byte)MathHelper.Clamp(alpha, Byte.MinValue, Byte.MaxValue);
+            if (((r | g | b | alpha) & 0xFFFFFF00) != 0)
+            {
+                R = (byte)MathHelper.Clamp(r, Byte.MinValue, Byte.MaxValue);
+                G = (byte)MathHelper.Clamp(g, Byte.MinValue, Byte.MaxValue);
+                B = (byte)MathHelper.Clamp(b, Byte.MinValue, Byte.MaxValue);
+                A = (byte)MathHelper.Clamp(alpha, Byte.MinValue, Byte.MaxValue);
+            }
+            else
+            {
+                R = (byte)r;
+                G = (byte)g;
+                B = (byte)b;
+                A = (byte)alpha;
+            }
+        }
+
+        /// <summary>
+        /// Constructs an RGBA color from scalars representing red, green, blue and alpha values.
+        /// </summary>
+        /// <remarks>
+        /// This overload sets the values directly without clamping, and may therefore be faster than the other overloads.
+        /// </remarks>
+        /// <param name="r"></param>
+        /// <param name="g"></param>
+        /// <param name="b"></param>
+        /// <param name="alpha"></param>
+        public Color(byte r, byte g, byte b, byte alpha)

```

Comment:
Do you mean this quote?

> When you pass integer values to a method that has byte and float overloads, C# method resolution rules choose the float version, because int -> float is considered a better conversion than int -> byte.

That is not a problem because we have the `int` overload, which is chosen by overload resolution since it has no conversions at all. The only way to get the `byte` or `float` overload is explicitly having all the parameters as the respective type (either by variable definition or casting).